pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -312,7 +312,16 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: _PATH) -> None:
             checkpoint: dict containing model and trainer state
             filepath: write-target file's path
         """
-        return self.checkpoint_io.save_checkpoint(checkpoint, filepath)
+        self.checkpoint_io.save_checkpoint(checkpoint, filepath)
+
+    def remove_checkpoint(self, filepath: _PATH) -> None:
+        """Remove checkpoint filepath from the filesystem.
+
+        Args:
+            filepath: Path to checkpoint
+        """
+        if self.local_rank == 0:
```

Comment:
This change is equivalent to what's in master. However, I don't understand why we don't check local rank for `save_checkpoint` but we do it for remove.

Reply:
because xm.save needs to be called on all ranks. It internally checks if the local rank is 0. This is just how the library works. If we don't call xm.save on all ranks, it will hang. 

https://pytorch.org/xla/release/1.5/_modules/torch_xla/core/xla_model.html

Reply:
Okay, I'll add a comment